test/gtest/core/test_mc.cc

```diff
@@ -82,7 +82,8 @@ UCC_TEST_F(test_mc, can_alloc_and_free_host_mem)
     ucc_mc_finalize();
 }
 
-UCC_TEST_F(test_mc, can_alloc_and_free_host_mem_mt)
+// Disabled because can't reinit mc with different thread mode
+UCC_TEST_F(test_mc, DISABLED_can_alloc_and_free_host_mem_mt)
```

Comment:
Why we can't re-init? What is the problem - finalizing mpools ? why can't we finalize them - i assume potential outstanding colls? @manjugv the question to you on a close  topic: do we alloc ucc_context_create being called while there is another outstanding UCC collective (on a team from another ucc context); i remember we don't allow concurrent ucc_ctx_create from multiple threads.

Reply:
it's up to memory component to do everything that needed to reinit correctly. I think we can implement it in one way or another, i just wanted to disable failing test in this pr and print warning to avoid debugging in future

Reply:
maybe we need to open gh issue then: 1. we need to understand if we need to support such re-init, 2. if we decide we need it we have to track so that we don't forget.

Reply:
sure, I think it's not P0 task since creating multiple ucc instances with different thread levels is something rare

Reply:
> Why we can't re-init? What is the problem - finalizing mpools ? why can't we finalize them - i assume potential outstanding colls? @manjugv the question to you on a close topic: do we alloc ucc_context_create being called while there is another outstanding UCC collective (on a team from another ucc context); i remember we don't allow concurrent ucc_ctx_create from multiple threads.

I think we should allow context create by multiple threads, when library is initialized the thread multiple mode. Given that it is a blocking call and can be local operation, is there a reason not to?

Reply:
The key here is "can be local" - you will have to specify exactly which combinations of params are allowed then for concurrent context creation. Obviously, it will not be allowed when OOB is provided.